modules/twinkleblock.js

```diff
@@ -0,0 +1,1064 @@
+	},
+	'uw-uhblock' : {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		reason: '{{uw-uhblock}} <!-- Username violation, hard block -->',
+		reasonParam: true,
+		summary: 'You have been indefinitely blocked from editing because your username is a blatant violation of the [[WP:U|username policy]]'
+	},
+	'uw-ublock-famous' : {
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		reason: '{{uw-ublock-famous}} <!-- Username represents a famous person, soft block -->',
+		summary: 'You have been indefinitely blocked from editing because your [[WP:U|username]] matches the name of a well-known living individual'
+	},
+	'uw-uhblock-double': {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		reason: '{{uw-ublock-double}} <!-- Username closely resembles another user, hard block -->',
+		summary: 'You have been indefinitely blocked from editing because your [[WP:U|username]] appears to impersonate another established Wikipedia user'
+	},
+	'uw-vaublock' : {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		pageParam: true,
+		reason: '{{uw-vaublock}} <!-- Username violation, vandalism-only account -->',
+		summary: 'You have been indefinitely blocked from editing because your account is being [[WP:VOA|used only for vandalism]] and your username is a blatant violation of the [[WP:U|username policy]]'
+	},
+	'uw-vblock' : {
+		autoblock: true,
+		expiry: '31 hours',
+		nocreate: true,
+		pageParam: true,
+		reason: '[[WP:Vandalism|Vandalism]]',
+		summary: 'You have been blocked from editing for persistent [[WP:VAND|vandalism]]'
+	},
+	'uw-voablock' : {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		pageParam: true,
+		reason: '[[WP:Vandalism-only account|Vandalism-only account]]',
+		summary: 'You have been indefinitely blocked from editing because your account is being [[WP:VOA|used only for vandalism]]'
+	}
+};
+
+Twinkle.block.transformBlockPresets = function twinkleblockTransformBlockPresets() {
+	// supply sensible defaults
+	$.each(Twinkle.block.blockPresetsInfo, function(preset, settings) {
+		settings.summary = settings.summary || settings.reason;
+		settings.expiry = settings.expiry || '31 hours';
+		settings.sig = settings.sig !== undefined ? settings.sig : 'yes';
+		// despite this it's preferred that you use 'infinity' as the value for expiry
+		settings.indefinite = settings.indefinite || settings.expiry === 'infinity' || settings.expiry === 'indefinite' || settings.expiry === 'never';
+		Twinkle.block.blockPresetsInfo[preset] = settings;
+	});
+};
+
+Twinkle.block.blockGroups = [
+	{
+		label: 'Common block reasons',
+		list: [
+			{ label: 'anonblock', value: 'anonblock' },
+			{ label: 'anonblock - likely a school', value: 'anonblock - school' },
+			{ label: 'school block', value: 'school block' },
+			{ label: 'Generic block', value: 'uw-block' },
```

Comment:
Maybe append `(custom reason)` to the description of the "Generic block" options to remind admins to type a custom block reason.